tests/test_consent/api/test_views.py

```diff
@@ -935,9 +983,45 @@ def test_consent_api_post_endpoint(self, factory, items, request_body,
             create_items(factory, items)
         response = self.client.post(self.path, request_body)
         self._assert_expectations(response, expected_response_body, expected_status_code)
-        # Assert that an enterprise course enrollment exists with consent provided.
-        if expected_status_code == 200:
-            self._assert_consent_provided(response)
+
+    @ddt.data(
+        (
+            factories.EnterpriseCustomerUserFactory,
+            [{
+                'user_id': TEST_USER_ID,
+                'enterprise_customer__uuid': TEST_UUID,
+                'enterprise_customer__enforce_data_sharing_consent': 'at_enrollment',
+            }],
+            {
+                DSCView.REQUIRED_PARAM_USERNAME: TEST_USERNAME,
+                DSCView.REQUIRED_PARAM_ENTERPRISE_CUSTOMER: TEST_UUID,
+                DSCView.REQUIRED_PARAM_COURSE_ID: TEST_COURSE,
+            },
+            {
+                DSCView.REQUIRED_PARAM_USERNAME: TEST_USERNAME,
+                DSCView.REQUIRED_PARAM_ENTERPRISE_CUSTOMER: TEST_UUID,
+                DSCView.REQUIRED_PARAM_COURSE_ID: TEST_COURSE,
+                DSCView.CONSENT_EXISTS: False,
+                DSCView.CONSENT_GRANTED: False,
+                DSCView.CONSENT_REQUIRED: False,
+            },
+            200
+        ),
+    )
+    @ddt.unpack
+    def test_consent_api_post_endpoint_course_not_in_catalog(
+            self,
+            factory,
+            items,
+            request_body,
+            expected_response_body,
+            expected_status_code
+    ):
+        self.cd_client.course_in_catalog.return_value = False
+        if factory:
```

Comment:
@haikuginger The data above doesn't seem to include a case where this would evaluate to `False`?